code/modules/hydroponics/grown/root.dm

```diff
@@ -28,10 +28,15 @@
 /obj/item/food/grown/carrot/attackby(obj/item/I, mob/user, params)
 	if(I.get_sharpness())
 		to_chat(user, span_notice("You sharpen the carrot into a shiv with [I]."))
-		var/obj/item/knife/shiv/carrot/Shiv = new /obj/item/knife/shiv/carrot
+		var/carrot_blade
+		var/carrot_sword_chance = 10 * (max(0, seed.potency - 50) / 50)
```

Comment:
How rare do you want this to be, actually? 

This is a 10% chance, which gives you a ~65% chance of getting a carrot sword in one harvest,

Reply:
In that case you may want to bump it down to a 1% chance (~9.5% chance per harvest)

Reply:
Done